flow/Net2.actor.cpp

```diff
@@ -462,6 +462,12 @@ class Connection : public IConnection, ReferenceCounted<Connection> {
 	}
 };
 
+#if BOOST_VERSION >= 107000
+#define GET_IO_SERVICE(s) ((boost::asio::io_context&)(s).get_executor().context())
+#else
+#define GET_IO_SERVICE(s) ((s).get_io_service())
+#endif
```

Comment:
I'm... not convinced that this is safe?  If we already have an io_context, then it seems better to use it, than reinterpret a tcp::acceptor as an io_context?  I think we already did some fixes here with #1312, and @vishesh might be able to help if he remembers the errors and changes that happened.

Reply:
thanks @alexmiller-apple this is taken from Jesse's original diff and I'm not able to refactor it. @vishesh would be great if you can help?